world-chain-builder/crates/world/payload/src/builder.rs

```diff
@@ -583,19 +587,14 @@ where
         let tx_da_limit = self.inner.da_config.max_da_tx_size();
         let base_fee = builder.evm_mut().block().basefee;
 
-        // Enable inspector for execution of the block transactions
-        builder.evm_mut().inspector_mut().active = true;
-
-        // TODO: perhaps we don't want to error out here.
-        let (builder_addr, stamp_block_tx) = crate::stamp::stamp_block_tx(self, builder.evm_mut())
-            .map_err(|e| PayloadBuilderError::Other(e.into()))?;
-
         let mut invalid_txs = vec![];
         let verified_gas_limit = (self.verified_blockspace_capacity as u64 * block_gas_limit) / 100;
 
+        let mut spent_nullifier_hashes = HashSet::new();
+
         // Subtract stamp_block_tx gas limit from the overall block gas limit.
         // This gas will be saved to execute stampBlock at the end of the block.
-        block_gas_limit -= stamp_block_tx.gas_limit();
+        block_gas_limit -= SPEND_NULLIFIER_HASHES_GAS;
```

Comment:
`SPEND_NULLIFIER_HASHES_GAS` will need to be a dynamic value since there will be `n` `SSTORE`s depending on how many PBH transactions are in the block.